### PR TITLE
Feature bn256 benchmark

### PR DESCRIPTION
**Pull request by plabanca**
Added benchmark tests for bn256

### CODE REVIEW

**Comment by spoonincode** [comment]
the bn256 CMake needs some attention:
https://github.com/AntelopeIO/bn256/blob/c3ae49fb432055650bcbaaa4d446e29e3562d7a5/src/CMakeLists.txt#L1
unnecessary since this is already specified at root CMakeLists.txt

https://github.com/AntelopeIO/bn256/blob/c3ae49fb432055650bcbaaa4d446e29e3562d7a5/src/CMakeLists.txt#L3
that should be removed

https://github.com/AntelopeIO/bn256/blob/c3ae49fb432055650bcbaaa4d446e29e3562d7a5/src/CMakeLists.txt#L4
Instead of using `${Boost_INCLUDE_DIRS}` should use `target_link_libraries(bn256 Boost::boost)`. However if that repo is intended for other projects to use, or to be built standalone, you really also need a `find_package(boost)` somewhere in the project (regardless of Boost_INCLUDE_DIRS or Boost::boost)

https://github.com/AntelopeIO/bn256/blob/c3ae49fb432055650bcbaaa4d446e29e3562d7a5/test/CMakeLists.txt#L2
should be removed

https://github.com/AntelopeIO/bn256/blob/c3ae49fb432055650bcbaaa4d446e29e3562d7a5/test/CMakeLists.txt#L14
LINK_PUBLIC is depreciated, there should be no need for it

https://github.com/AntelopeIO/bn256/blob/c3ae49fb432055650bcbaaa4d446e29e3562d7a5/test/CMakeLists.txt#L6
this line shouldn't be needed at all once you convert to Boost::boost usage and get rid of LINK_PUBLIC

https://github.com/AntelopeIO/bn256/blob/c3ae49fb432055650bcbaaa4d446e29e3562d7a5/test/CMakeLists.txt#L24
the TARGET_FILE stuff is not required

File: benchmark/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 file(GLOB BENCHMARK "*.cpp")
-add_executable( benchmark ${BENCHMARK} )
+add_executable( benchmark ${BENCHMARK} bn_256.cpp)
```
Comment (spoonincode) [inline benchmark/CMakeLists.txt:2]: Unneeded -- `BENCHMARK` is a glob of all cpp files

File: .gitmodules [hunk 2]
```diff
@@ -28,3 +28,6 @@
 [submodule "libraries/libfc/libraries/ff"]
 	path = libraries/libfc/libraries/ff
 	url = https://github.com/AntelopeIO/libff
+[submodule "libraries/libfc/libraries/bn256"]
+	path = libraries/libfc/libraries/bn256
+	url = https://github.com/AntelopeIO/bn256.git
```
Comment (spoonincode) [inline .gitmodules:33]: All of the other submodules are without `.git` at the end. It doesn't affect anything, but for consistency consider making it like the others.

Comment (plabanca) [reply .gitmodules]: resolve with https://github.com/AntelopeIO/leap/pull/280/commits/1404bb83e7c81bc50a2d4b0c7ba70b787870badc

File: benchmark/CMakeLists.txt (hunk 1 above)
Comment (plabanca) [reply benchmark/CMakeLists.txt]: resolve with https://github.com/AntelopeIO/leap/pull/280/commits/f1f96883e1bae16a13161a456f201695e91f6498

**Pull request closed without merging**